src/main/java/com/mixpanel/android/viewcrawler/EditProtocol.java

```diff
@@ -90,35 +90,53 @@ public ViewVisitor readEdit(JSONObject source) throws BadInstructionsException {
-                targetClass = Class.forName(targetClassName);
-            } catch (final ClassNotFoundException e) {
-                throw new BadInstructionsException("Can't find class for visit path: " + targetClassName, e);
-            }
+                final Class<?> targetClass;
+                try {
+                    targetClass = Class.forName(targetClassName);
+                } catch (final ClassNotFoundException e) {
+                    throw new BadInstructionsException("Can't find class for visit path: " + targetClassName, e);
+                }
 
-            final PropertyDescription prop = readPropertyDescription(targetClass, source.getJSONObject("property"));
-            final JSONArray argsAndTypes = source.getJSONArray("args");
-            final Object[] methodArgs = new Object[argsAndTypes.length()];
-            for (int i = 0; i < argsAndTypes.length(); i++) {
-                final JSONArray argPlusType = argsAndTypes.getJSONArray(i);
-                final Object jsonArg = argPlusType.get(0);
-                final String argType = argPlusType.getString(1);
-                methodArgs[i] = convertArgument(jsonArg, argType);
-            }
+                final PropertyDescription prop = readPropertyDescription(targetClass, source.getJSONObject("property"));
+                final JSONArray argsAndTypes = source.getJSONArray("args");
+                final Object[] methodArgs = new Object[argsAndTypes.length()];
+                for (int i = 0; i < argsAndTypes.length(); i++) {
+                    final JSONArray argPlusType = argsAndTypes.getJSONArray(i);
+                    final Object jsonArg = argPlusType.get(0);
+                    final String argType = argPlusType.getString(1);
+                    methodArgs[i] = convertArgument(jsonArg, argType);
+                }
 
-            final Caller mutator = prop.makeMutator(methodArgs);
-            if (null == mutator) {
-                throw new BadInstructionsException("Can't update a read-only property " + prop.name + " (add a mutator to make this work)");
-            }
+                final Caller mutator = prop.makeMutator(methodArgs);
+                if (null == mutator) {
+                    throw new BadInstructionsException("Can't update a read-only property " + prop.name + " (add a mutator to make this work)");
+                }
+
+                return new ViewVisitor.PropertySetVisitor(path, mutator, prop.accessor);
+            } else if (source.getString("change_type").equals("layout")) {
+                final JSONArray args = source.getJSONArray("args");
+                JSONObject layout_info = args.optJSONObject(0);
+                ViewVisitor.LayoutRule params;
+                int verb = layout_info.getInt("verb");
+                if (layout_info.getString("operation").equals("remove")) {
+                    params = new ViewVisitor.LayoutRule(verb, 0);
+                } else if (layout_info.has("anchor_id")) {
```

Comment:
Could input ever have both anchor_id and remove? I feel like we might be able to streamline this, so that we can't create invalid combinations of properties. Perhaps just expect anchor_id == 0?